src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/Everything/PatientEverythingService.cs

```diff
@@ -0,0 +1,369 @@
+
+            // Will enable this after more tests
+            if (string.Equals(search.Value.GetType().Name, "SqlServerSearchService", StringComparison.Ordinal))
+            {
+                throw new OperationNotImplementedException("$everything operation is not yet implemented in SQL Server.");
+            }
+
+            EverythingOperationContinuationToken token = string.IsNullOrEmpty(continuationToken)
+                ? new EverythingOperationContinuationToken(0, null)
+                : EverythingOperationContinuationToken.FromString(DecodeContinuationTokenFromBase64String(continuationToken));
+
+            if (token == null)
+            {
+                throw new BadRequestException(Core.Resources.InvalidContinuationToken);
+            }
+
+            SearchResult searchResult;
+            string encodedInternalContinuationToken = EncodeContinuationToken(token.InternalContinuationToken);
+
+            switch (token.Phase)
+            {
+                case 0:
+                    searchResult = await SearchIncludes(resourceType, resourceId, since, type, cancellationToken);
+                    if (!searchResult.Results.Any())
+                    {
+                        token.Phase = 1;
+                        goto case 1;
+                    }
+
+                    break;
+                case 1:
+                    // If both start and end are null, we can just perform regular compartment search in Phase 2
+                    if (start == null && end == null)
+                    {
+                        token.Phase = 2;
+                        goto case 2;
+                    }
+
+                    searchResult = await SearchCompartmentWithDate(resourceType, resourceId, start, end, since, type, encodedInternalContinuationToken, cancellationToken);
+                    if (!searchResult.Results.Any())
+                    {
+                        token.Phase = 2;
+                        goto case 2;
+                    }
+
+                    break;
+                case 2:
+                    searchResult = start == null && end == null
+                        ? await SearchCompartment(resourceType, resourceId, since, type, encodedInternalContinuationToken, cancellationToken)
+                        : await SearchCompartmentWithoutDate(resourceType, resourceId, since, type, encodedInternalContinuationToken, cancellationToken);
+
+                    if (!searchResult.Results.Any())
+                    {
+                        token.Phase = 3;
+                        goto case 3;
+                    }
+
+                    break;
+                case 3:
+                    searchResult = await SearchRevinclude(resourceId, since, type, encodedInternalContinuationToken, cancellationToken);
+                    break;
+                default:
+                    return new SearchResult(Enumerable.Empty<SearchResultEntry>(), null, null, Array.Empty<Tuple<string, string>>());
```

Comment:
>return new SearchResult(Enumerable.Empty<SearchResultEntry>(), null, null, Array.Empty<Tuple<string, string>>()); [](http://example.com/codeflow?start=20&length=113)

I would throw exception.
This shouldn't happen in theory, so it sounds like exceptional situation for me.

Reply:
Agree. It is refined in the update. If this happens, a `EverythingOperationException` would be thrown and return `InternalServerError`.